src/components/Tooltip/Tooltip.tsx

```diff
@@ -56,113 +56,221 @@ const paddingsBottom = {
+            case currentPlacement.toString().includes(TooltipPosition.Left):
+                return merge([
+                    "before:tw-border-b-0 before:tw-border-l-0 tw-right-[-6px]",
+                    brightHeader && alignment === TooltipAlignment.Start
+                        ? brightHeaderArrowBackgroundColors[brightHeader]
+                        : "before:tw-bg-black-100 before:dark:tw-bg-white",
+                ]);
+            default:
+                return "before:tw-border-b-0 before:tw-border-r-0 tw-top-[-6px] before:tw-bg-black-100 before:dark:tw-bg-white";
+        }
+    };
+
+    const tooltipOffset = withArrow ? 10 : 5;
+    const popperInstance = usePopper(triggerRefElement?.current, tooltipContainerRef.current, {
+        // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+        // @ts-ignore
+        placement: placement,
+        modifiers: [
+            {
+                name: "arrow",
+                options: {
+                    element: arrowElement,
+                },
+            },
+            {
+                name: "offset",
+                options: {
+                    offset: [0, tooltipOffset],
+                },
+            },
+            {
+                name: "flip",
+                enabled: flip,
+            },
+        ],
+    });
+
+    const arrowStyling = setArrowClasses();
+    const [isOpen, setIsOpen] = useState(false);
+
+    useEffect(() => {
+        if (triggerRefElement) {
+            triggerRefElement.current.addEventListener("click", () => setIsOpen(!isOpen));
+            triggerRefElement.current.addEventListener("mouseover", () => setIsOpen(true));
+            triggerRefElement.current.addEventListener("mouseleave", () => setTimeout(() => setIsOpen(false), 2000));
+            triggerRefElement.current.addEventListener("focus", () => setIsOpen(true));
+        }
+
+        return () => {
+            if (triggerRefElement) {
+                triggerRefElement.current.removeEventListener("click", () => setIsOpen(!isOpen));
+                triggerRefElement.current.removeEventListener("mouseover", () => setIsOpen(true));
+                triggerRefElement.current.removeEventListener("mouseleave", () =>
+                    setTimeout(() => setIsOpen(false), 2000),
+                );
+                triggerRefElement.current.removeEventListener("focus", () => setIsOpen(true));
+            }
+        };
+    }, []);
 
     return (
-        <>
-            <div
-                ref={tooltipContainerRef}
-                className="tw-popper-container tw-inline-block tw-max-w-[200px] tw-bg-black-100 dark:tw-bg-white tw-rounded-md tw-shadow-mid tw-text-white dark:tw-text-black-100 tw-z-20"
-                style={style}
-                data-test-id="tooltip"
-                {...popperAttributes}
-                {...tooltipProps}
-            >
-                {brightHeader && <BrightHeader headerStyle={brightHeader} />}
+        <div>
```

Comment:
Maybe we could directly use ReactPortal here to avoid any positioning issues with other elements on the page.

Reply:
Agreed. Both me and Lenny currently have to use Portal to correctly position the Tooltip, would be great if this was built into the component itself.

Reply:
@lenardpalko @Kenny806 I just updated according to all comments. I agree it makes sense to use Portal within the component. I also fixed the timeout issue.

Reply:
@imoutaharik From my side it works fine now. I won't approve the PR since @thisguyscode still has some comments, but the way Tooltip is used is good for me and I have no further comments there.